cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -386,174 +369,4 @@ protected boolean respondIfElementNotFound(Throwable t, HttpResponder responder)
     }
     return false;
   }
-
-  // TODO: refactor
-  protected final void dataList(HttpResponder responder, Store store, DatasetFramework dsFramework,
```

Comment:
Correct. It was used before, but then this functionality is replaced by the Datasets Registry concept.
You can see usages of it being removed here, but the method itself was not removed.
https://github.com/caskdata/cdap/pull/2166/files#diff-2b14ed88e3eb39a9c4550dfd17f60b8aL80